### PR TITLE
Feature uart formats

### PR DESCRIPTION
**Pull request by rafacouto**
Reference issue #132

### CODE REVIEW

File: src/drivers/ports/esp32/uart_port.h [hunk 1]
```diff
@@ -33,6 +33,46 @@
...
+#define UART_FORMAT_7O15  (ESP32_UART_DATA_BITS_7 | ESP32_UART_PARITY_O | ESP32_UART_STOP_BITS_15)
+#define UART_FORMAT_7E1   (ESP32_UART_DATA_BITS_7 | ESP32_UART_PARITY_E | ESP32_UART_STOP_BITS_1)
+#define UART_FORMAT_7E15  (ESP32_UART_DATA_BITS_7 | ESP32_UART_PARITY_E | ESP32_UART_STOP_BITS_15)
+#define UART_FORMAT_8N1   (ESP32_UART_DATA_BITS_8 | ESP32_UART_PARITY_N | ESP32_UART_STOP_BITS_1)
+#define UART_FORMAT_8N15  (ESP32_UART_DATA_BITS_8 | ESP32_UART_PARITY_N | ESP32_UART_STOP_BITS_15)
+#define UART_FORMAT_8O1   (ESP32_UART_DATA_BITS_8 | ESP32_UART_PARITY_O | ESP32_UART_STOP_BITS_1)
+#define UART_FORMAT_8O15  (ESP32_UART_DATA_BITS_8 | ESP32_UART_PARITY_O | ESP32_UART_STOP_BITS_15)
+#define UART_FORMAT_8E1   (ESP32_UART_DATA_BITS_8 | ESP32_UART_PARITY_E | ESP32_UART_STOP_BITS_1)
+#define UART_FORMAT_8E15  (ESP32_UART_DATA_BITS_8 | ESP32_UART_PARITY_E | ESP32_UART_STOP_BITS_15)
+#define UART_FORMAT_DEFAULT     UART_FORMAT_8N1
```
Comment (rafacouto) [inline src/drivers/ports/esp32/uart_port.h]: But uart_init() is port-agnostic and a port is initializated with its default format. Defining UART_FORMAT_DEFAULT means that drivers know the format to their implementations.

Comment (eerimoq) [inline src/drivers/ports/esp32/uart_port.h]: Fine, we can keep it like this, but I find it highly unlikely that any port will have a default format not 8N1 =)

Long term there should be a port-specific part of ``struct uart_driver_t``, so ports can choose to ignore format all together.

**Comment by rafacouto** [comment]
@eerimoq, thank you for your comments 👍 Now, it is compiling and WIP is pointing to study datasheets and to define UART_PORT_FORMAT_* in each port. 😸

**Comment by eerimoq** [comment]
Great!

One more comment, would be nice if the format defines and the set function share a common name, that is, either change the defines to ``UART_FRAME_FORMAT_``, or change the function name back to ``uart_set_format()``. I think frame format is the correct naming, but I could be wrong.

**Comment by rafacouto** [comment]
I have rebased from current master in PR #140

**Pull request closed without merging**